src/Aspire.Hosting.SqlServer/SqlServerBuilderExtensions.cs

```diff
@@ -27,7 +27,8 @@ public static IResourceBuilder<SqlServerServerResource> AddSqlServer(this IDistr
         var sqlServer = new SqlServerServerResource(name, passwordParameter);
         return builder.AddResource(sqlServer)
                       .WithEndpoint(port: port, targetPort: 1433, name: SqlServerServerResource.PrimaryEndpointName)
-                      .WithImage(SqlServerContainerImageTags.Image, SqlServerContainerImageTags.Tag)
+                      .WithImage(SqlServerContainerImageTags.Image)
+                      .WithImageSHA256(SqlServerContainerImageTags.Digest)
```

Comment:
> they suggested we should instead depend on digests as opposed to tags, to be able to guarantee determinism

Note that this is OK to unblock our tests, but this doesn't comply with our long term plan.

https://github.com/dotnet/aspire/issues/3933
https://github.com/dotnet/aspire/issues/1997#issuecomment-1924592452

If

Reply:
Let's have a follow up discussion just to finalize on our plan going forward. We may very well end up following the same plan we have, I just want to make sure we take into consideration the concept of determinism and the fact that we have now been broken by this, so we should check what our customers would expect us to do.

For this particular case, it's worth noting that the underlying image doesn't follow semver  which seems to be one of the keys of our plan.